BitList.cs

```diff
@@ -0,0 +1,1552 @@
+    public bool Xor()
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+        {
+            bool result = bits[0];
+            for (int i = 1; i < bits.Length; i++)
+                result ^= bits[i];
+            return result;
+        }
+    }
+
+    /// <summary>
+    /// XOR boolean operation with another list of bit.
+    /// </summary>
+    /// <exception cref="BooleanAlgebraException" />
+    public void Xor(BitList bits)
+    {
+        if (IsEmpty)
+            throw new BooleanAlgebraException();
+        else
+            this.bits.Xor(bits.bits);
+    }
+
+    /// <summary>
+    /// Add a list of bit to the last position of this instance.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public void Add(BitList bits)
+    {
+        var newBitList = IsEmpty ? bits : this + bits;
+        this.bits = newBitList.bits;
+    }
+
+    /// <summary>
+    /// Add a bit to the last position of this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public void Add(bool bit)
+    {
+        var newBitList = IsEmpty ? new BitList(bit) : this + bit;
+        bits = newBitList.bits;
+    }
+
+    /// <summary>
+    /// Add a list of bit to the first position of this instance.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public void AddFirst(BitList bits)
+    {
+        Insert(0, bits);
+    }
+
+    /// <summary>
+    /// Add a bit to the first position of this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public void AddFirst(bool bit)
+    {
+        Insert(0, bit);
+    }
+
+    /// <summary>
+    /// Remove a bit in the specified position.
+    /// </summary>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void RemoveAt(int index)
+    {
+        if (!IsEmpty && index < Count && index >= 0)
+        {
+            var bools = ToBooleanArray() as IEnumerable<bool> as List<bool>;
+            bools.RemoveAt(index);
+            bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new ArgumentOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Remove a range of bits from this instance.
+    /// </summary>
+    /// <param name="index">Start index.</param>
+    /// <param name="count">Count of bits.</param>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public void RemoveRange(int index, int count)
+    {
+        if (!IsEmpty && index < bits.Length && index >= 0 && count <= bits.Length - index && count > 0)
+        {
+            var bools = ToBooleanArray() as IEnumerable<bool> as List<bool>;
+            bools.RemoveRange(index, count);
+            bits = new BitArray(bools.ToArray());
+        }
+        else
+            throw new ArgumentOutOfRangeException();
+    }
+
+    /// <summary>
+    /// Remove the specified bit from this instance.
+    /// </summary>
+    public bool Remove(bool bit)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename parameter 'bit' to 'item' to match the interface declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)